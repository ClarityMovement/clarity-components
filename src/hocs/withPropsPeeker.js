import withProps from 'recompose/withProps';

export default object => withProps(props => Object.assign(object, props));
