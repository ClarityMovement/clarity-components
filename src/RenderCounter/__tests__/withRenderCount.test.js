/* eslint-env jest */
import React from 'react';
import T from 'prop-types';
import { mount } from 'enzyme';
import compose from 'recompose/compose';
import { withRenderCount } from '../';
import withPropsPeeker from '../../hocs/withPropsPeeker';

describe('withRenderCount(BaseComponent): NewComponent', () => {
  describe('modifies props', () => {
    let BaseComponent;
    let props;
    let NewComponent;

    beforeEach(() => {
      props = {};
      BaseComponent = () => <div />;

      NewComponent = compose(
        withRenderCount,
        withPropsPeeker(props),
      )(BaseComponent);
    });

    it('should transfer initialCount as initial value of prop count', () => {
      mount(<NewComponent initialCount={2} />);
      expect(props).toEqual({ count: 2 });
    });

    it('should +1 count prop after component update', () => {
      const wrapper = mount(<NewComponent />);

      expect(props).toEqual({ count: 1 });
      wrapper.update();
      expect(props).toEqual({ count: 2 });
    });
  });

  test('NewComponent has the extended statics of BaseComponent', () => {
    const BaseComponent = () => <div />;
    BaseComponent.propTypes = {
      count: T.number,
      foo: T.string,
    };
    BaseComponent.defaultProps = { count: 1 };

    const NewComponent = withRenderCount(BaseComponent);

    expect(NewComponent).not.toBe(BaseComponent);
    expect(NewComponent.displayName).toBe('withRenderCount(BaseComponent)');
    expect(NewComponent.propTypes).toEqual({
      initialCount: T.number,
      foo: T.string,
    });
    expect(NewComponent.defaultProps.initialCount).toBe(1);
  });
});
