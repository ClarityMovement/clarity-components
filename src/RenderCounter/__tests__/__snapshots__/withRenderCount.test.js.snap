// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRenderCount(BaseComponent): NewComponent NewComponent has the extended statics of BaseComponent 1`] = `
Object {
  "defaultProps": Object {
    "initialCount": 1,
  },
  "displayName": "withRenderCount(BaseComponent)",
  "propTypes": Object {
    "foo": [Function],
    "initialCount": [Function],
    "onRerender": [Function],
  },
}
`;

exports[`withRenderCount(BaseComponent): NewComponent modifies props should transfer initialCount as initial value of prop count 1`] = `
<div
  name="props-displayer"
>
  <input-props
    initialCount={2}
  />
  <output-props
    count={2}
  />
</div>
`;
